drivers/interrupt_controller/intc_mchp_ecia_xec.c

```diff
@@ -0,0 +1,1135 @@
+	    (src < 0) || (src > 31)) {
+		return -EINVAL;
+	}
+
+	struct ecia_regs *regs = ECIA_XEC_REG_BASE;
+
+	/* write 1 to clear */
+	regs->GIRQ[girq - MCHP_FIRST_GIRQ].EN_CLR = BIT(src);
+
+	return 0;
+}
+
+/* forward reference */
+static const struct device *get_girq_dev(int girq_num);
+
+/**
+ * @brief set GIRQn interrupt source callback
+ *
+ * @param dev_girq is the GIRQn device handle
+ * @param src is the interrupt source in the GIRQ (0 - 31)
+ * @param cb user callback
+ * @param data user data
+ */
+int mchp_xec_ecia_set_callback_by_dev(const struct device *dev_girq, int src,
+				      mchp_xec_ecia_callback_t cb, void *data)
+{
+	if ((dev_girq == NULL) || (src < 0) || (src > 31)) {
+		return -EINVAL;
+	}
+
+	const struct xec_girq_config *const cfg = DEV_GIRQ_CFG(dev_girq);
+	struct xec_girq_src_data *girq_data = DEV_GIRQ_DATA(dev_girq);
+
+	/* source exists in this GIRQ? */
+	if (!(cfg->sources[src] & BIT(7))) {
+		return -EINVAL;
+	}
+
+	/* obtain the callback array index for the source */
+	int idx = (int)(cfg->sources[src] & ~BIT(7));
+
+	girq_data[idx].cb = cb;
+	girq_data[idx].data = data;
+
+	return 0;
+}
+
+/**
+ * @brief set GIRQn interrupt source callback
+ *
+ * @param girq is the GIRQ number (8 - 26)
+ * @param src is the interrupt source in the GIRQ (0 - 31)
+ * @param cb user callback
+ * @param data user data
+ */
+int mchp_xec_ecia_set_callback(int girq_num, int src,
+			       mchp_xec_ecia_callback_t cb, void *data)
+{
+	const struct device *dev = get_girq_dev(girq_num);
+
+	return mchp_xec_ecia_set_callback_by_dev(dev, src, cb, data);
+}
+
+/**
+ * @brief unset GIRQn interrupt source callback by device handle
+ *
+ * @param dev_girq is the GIRQn device handle
+ * @param src is the interrupt source in the GIRQ (0 - 31)
+ */
+int mchp_ecia_unset_callback_by_dev(const struct device *dev_girq, int src)
+{
+	if ((dev_girq == NULL) || (src < 0) || (src > 31)) {
+		return -EINVAL;
+	}
+
+	const struct xec_girq_config *const cfg = DEV_GIRQ_CFG(dev_girq);
+	struct xec_girq_src_data *girq_data = DEV_GIRQ_DATA(dev_girq);
+
+	/* source exists in this GIRQ? */
+	if (!(cfg->sources[src] & BIT(7))) {
+		return -EINVAL;
+	}
+
+	/* obtain the callback array index for the source */
+	int idx = (int)(cfg->sources[src] & ~BIT(7));
+
+	girq_data[idx].cb = NULL;
+	girq_data[idx].data = NULL;
+
+	return 0;
+}
+
+/**
+ * @brief unset GIRQn interrupt source callback
+ *
+ * @param girq is the GIRQ number (8 - 26)
+ * @param src is the interrupt source in the GIRQ (0 - 31)
+ */
+int mchp_ecia_unset_callback(int girq_num, int src)
+{
+	const struct device *dev = get_girq_dev(girq_num);
+
+	return mchp_ecia_unset_callback_by_dev(dev, src);
+}
+
+#if (GIRQ08_AGGR || GIRQ09_AGGR || GIRQ10_AGGR || GIRQ11_AGGR ||\
+	GIRQ12_AGGR || GIRQ12_AGGR || GIRQ13_AGGR || GIRQ14_AGGR ||\
+	GIRQ15_AGGR || GIRQ16_AGGR || GIRQ17_AGGR || GIRQ18_AGGR ||\
+	GIRQ19_AGGR || GIRQ20_AGGR || GIRQ21_AGGR || GIRQ23_AGGR ||\
+	GIRQ24_AGGR || GIRQ25_AGGR || GIRQ26_AGGR)
```

Comment:
this conditional is unwieldy and too large to maintain, please fix

Reply:
Please advise.  Do you want a another macro containing this macro?  The ISR is required any any of the nodes is enabled. No required if no node is enabled.  Without the conditional the compile complains of unused function when all the GIRQ nodes are disabled.

Reply:
Found a way to do this with DT_FOREACH removing all the GIRQxx_AGGR logic.